components/Header/src/HeaderLogic.tsx

```diff
@@ -0,0 +1,247 @@
+      setMobileMenuActive(false);
+      if (lastDesktopMenuActive === menuConstants.LANGUAGE) {
+        setLanguageSwitcherActive(true);
+      } else {
+        setWelcomeMenuActive(true);
+      }
+    } else if ((welcomeMenuActive || languageSwitcherActive) && windowWidth < menuConstants.BREAKPOINT) {
+      setWelcomeMenuActive(false);
+      setLanguageSwitcherActive(false);
+      setMobileMenuActive(true);
+    }
+  }, [windowWidth]);
+
+  const closeButtonRef = useRef<HTMLButtonElement>(null);
+  const menuButtonRef = useRef<HTMLButtonElement>(null);
+
+  useEffect(() => {
+    mobileMenuDialogRef.current?.showModal();
+    if (mobileMenuActive) {
+      closeButtonRef.current?.focus();
+    } else if (mobileMenuActivatedBefore) {
+      menuButtonRef.current?.focus();
+    }
+  }, [mobileMenuActive]);
+
+  return (
+    <Header>
+      <ResponsiveContent>
+        <HeaderContent>
+          <HeaderLogoContainer>
+            <Link
+              href="https://www.denhaag.nl"
+              className="denhaag-logo denhaag-header__link"
+              aria-label="Gemeente Den Haag"
+            >
+              <HeaderLogo />
+            </Link>
+          </HeaderLogoContainer>
+          <HeaderActions>
+            <HeaderAction className="denhaag-header__actions-action-language-switcher">
+              <MenuButtonExpandable active={languageSwitcherActive} onClick={handleLanguageSwitcherToggle}>
+                NL
+              </MenuButtonExpandable>
+            </HeaderAction>
+            <HeaderAction className="denhaag-header__actions-action-user-menu">
+              <MenuButtonExpandable active={welcomeMenuActive} onClick={handleWelcomeMenuToggle}>
+                {props.userprofileMenu.label}
+              </MenuButtonExpandable>
+            </HeaderAction>
+          </HeaderActions>
+          <HeaderMobileActions>
+            <Button
+              className="denhaag-button denhaag-button--primary-action"
+              onClick={handleMobileMenuToggle}
+              ref={menuButtonRef}
+            >
+              Menu
+            </Button>
+          </HeaderMobileActions>
+        </HeaderContent>
+      </ResponsiveContent>
+      <Breadcrumb navigationPath={props.breadcrumbs} />
+      {languageSwitcherActive && (
+        <React.Fragment>
+          <Sheet>
+            <ResponsiveContent>
+              <IconButton className="denhaag-sheet__close-button" onClick={handleLanguageSwitcherToggle}>
+                <CloseIcon />
+              </IconButton>
+              <SheetContainer>
+                <Heading4>Kies uw voorkeurstaal</Heading4>
+                <LanguageSwitcherLogic {...props.languageSwitcherMenu}></LanguageSwitcherLogic>
+              </SheetContainer>
+            </ResponsiveContent>
+          </Sheet>
+          <SheetOverlay onClick={handleCloseSheet} />
+        </React.Fragment>
+      )}
+      {welcomeMenuActive && (
+        <React.Fragment>
```

Comment:
**Side node**: In other projects I use `<></>` instead of `<React.Fragment>` maybe we can discuss which you prefer and change them in the codebase in a separate PR.

Reply:
I prefer `<></>`, i'm going to change this.